music21/stream/core.py

```diff
@@ -191,9 +253,8 @@ def coreElementsChanged(
                 'coreElementsChanged should not be triggered on an immutable stream'
             )
 
-        if memo is None:
-            memo = []
-        memo.append(id(self))
+        if memo is not None:
+            memo.append(id(self))
```

Comment:
Actually, leave this... but should put after the `if` statement, `if id(self) in memo: return` and...

Reply:
in line `livingSite.coreElementsChanged()` we should be passing in the memo at least and probably eventually `updateIsFlat` and `clearIsSorted` -- this should be a significant speedup.

Reply:
Wasn't sure if "leave this" meant you wanted memo.append(id(self)) to still run even when memo is None. Currently we do nothing with the memo.